File: app/src/main/java/soup/movie/ui/theater/TheaterEditViewState.java
package soup.movie.ui.theater;

import java.util.List;

import soup.movie.data.model.TheaterCode;
import soup.movie.ui.BaseViewState;

class TheaterEditViewState implements BaseViewState {

    private final List<TheaterCode> allTheaters;
    private final List<TheaterCode> selectedTheaters;

    public TheaterEditViewState(List<TheaterCode> allTheaters, List<TheaterCode> selectedTheaters) {
        this.allTheaters = allTheaters;
        this.selectedTheaters = selectedTheaters;
    }

    public List<TheaterCode> getAllTheaters() {
        return allTheaters;
    }

    public List<TheaterCode> getSelectedTheaters() {
        return selectedTheaters;
    }

    @Override
    public String toString() {
        return "TheaterEditViewState{" +
                "allTheaters=" + allTheaters +
                ", selectedTheaters=" + selectedTheaters +
                '}';
    }
}

File: app/src/main/java/soup/movie/ui/theater/edit/TheaterEditViewState.kt
package soup.movie.ui.theater.edit

import androidx.annotation.Keep
import soup.movie.data.TheaterEditManager.Companion.MAX_ITEMS

sealed class TheaterEditViewState {

    @Keep
    object LoadingState : TheaterEditViewState() {

        override fun toString(): String = javaClass.simpleName
    }

    @Keep
    object ErrorState : TheaterEditViewState() {

        override fun toString(): String = javaClass.simpleName
    }

    @Keep
    data class DoneState(
            val selectedItemCount: Int) : TheaterEditViewState()

    fun getCurrentCount(): Int =
            when (this) {
                is DoneState -> selectedItemCount
                else -> 0
            }

    fun isFull(): Boolean = getCurrentCount() >= MAX_ITEMS
}
